staging/src/k8s.io/component-base/metrics/opts_test.go

```diff
@@ -138,3 +140,68 @@ func TestConstrainLabelMap(t *testing.T) {
 		})
 	}
 }
+
+func TestSetLabelAllowListFromManifest(t *testing.T) {
+	tests := []struct {
+		name                       string
+		manifest                   string
+		manifestExist              bool
+		expectlabelValueAllowLists map[string]*MetricLabelAllowList
+	}{
+		{
+			name:          "successfully parse manifest",
+			manifestExist: true,
+			manifest: `metric1,label1: v1,v2
+metric2,label2: v3`,
+			expectlabelValueAllowLists: map[string]*MetricLabelAllowList{
+				"metric1": {
+					labelToAllowList: map[string]sets.String{
+						"label1": sets.NewString("v1", "v2"),
+					},
+				},
+				"metric2": {
+					labelToAllowList: map[string]sets.String{
+						"label2": sets.NewString("v3"),
+					},
+				},
+			},
+		},
+		{
+			name:                       "failed to read manifest file",
+			manifestExist:              false,
+			expectlabelValueAllowLists: map[string]*MetricLabelAllowList{},
+		},
+		{
+			name:          "failed to parse manifest",
+			manifestExist: true,
+			manifest: `allow-list:
+- metric1,label1:v1
+- metric2,label2:v2,v3`,
+			expectlabelValueAllowLists: map[string]*MetricLabelAllowList{},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			labelValueAllowLists = map[string]*MetricLabelAllowList{}
+			manifestFilePath := "/non-existent-file.yaml"
+			if tc.manifestExist {
+				tempFile, err := os.CreateTemp("", "allow-list-test")
+				if err != nil {
+					t.Fatalf("failed to create temp file: %v", err)
+				}
+				defer os.Remove(tempFile.Name())
+
+				if _, err := tempFile.WriteString(tc.manifest); err != nil {
+					t.Fatalf("failed to write to temp file: %v", err)
+				}
+				manifestFilePath = tempFile.Name()
+			}
+
+			SetLabelAllowListFromManifest(manifestFilePath)
+			if !cmp.Equal(labelValueAllowLists, tc.expectlabelValueAllowLists, cmp.AllowUnexported(MetricLabelAllowList{})) {
```

Comment:
```suggestion
			if !reflect.DeepEqual(labelValueAllowLists, tc.expectlabelValueAllowLists) {
```

Reply:
I recommend we use `cmp.Equal` here as [cmp package](https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/google/go-cmp/cmp/compare.go#L7) is more powerful alternative to `reflect.DeepEqual`

Reply:
In the context of the operation being done here, I believe the two fulfill the same purpose, however, as a thumb rule, I'd prefer standard library over an external package, wherever possible.

Reply:
I see. To be consistent with the existing test case,  I've updated to use `reflect.DeepEqual`.